File: 3658 GCD of Odd and Even Sums/leet_better.js
/**
 * @param {number} n
 * @return {number}
 */
var gcdOfOddEvenSums = function(n) {
    const gcd = (x, y) => {
        while (y !== 0) {
            [x, y] = [y, x % y]
        }
        return x
    }
    const sumOdd = n * n
    const sumEven = 2 * n * n + n

    return gcd(sumOdd, sumEven)
};

File: 3658 GCD of Odd and Even Sums/leet_better.rs
impl Solution {
    pub fn gcd_of_odd_even_sums(n: i32) -> i32 {
        fn gcd(mut a: i32, mut b: i32) -> i32 {
            while b != 0 {
                (a, b) = (b, a % b)
            }

            a
        }

        let sumEven = n * n;
        let sumOdd = 2 * n * n + n;

        return gcd(sumEven, sumOdd);
    }
}